### PR TITLE
Feature: Implement automated rating request

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Closes https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/370.

This PR implements automated rating request for the AppStore. 

1st attempt: It will show the rating popup each 10th start/wakeup.

## Summary for release notes

Feature: Implement automated rating request

### CODE REVIEW

**Comment by wutschel** [comment]
I added some logic which will save the current version in the key `"VersionUnderReview"` each time the version is updated. If the current version is different to the one under review, the counter will be reset to 0. If the version keeps same, the counter in key `"PlaybackAttempts"` is increased and each 50th time the review request will pop up. I am not 100% sure after how many attempts the popup shall appear. I am also thinking of adding a check for a certain amount of time to be passed before we ask again (e.g. 6 months or so). 

What are your thoughts?

**Comment by wutschel** [comment]
@kambala-decapitator, do you see this is a good choice to trigger the reviews? And if so, should we add this for the next release already?

File: XBMC Remote/Utilities.m [hunk 1]
```diff
@@ -859,13 +859,25 @@ + (void)showReviewController {
...
 
 + (void)checkForReviewRequest {
-    // Read and increase count by 1
-    NSInteger count = [[NSUserDefaults standardUserDefaults] integerForKey:@"PlayCount"] + 1;
-    [[NSUserDefaults standardUserDefaults] setInteger:count forKey:@"PlayCount"];
-    
-    // Show review popup each 10th launch/wakeup
-    if (count % 10 == 0) {
-        [Utilities showReviewController];
+    id infoDict = NSBundle.mainBundle.infoDictionary;
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:862]: why `id`? The type is known. Please either use `__auto_type` or `NSDictionary*`.

File: XBMC Remote/Utilities.m [hunk 2]
```diff
@@ -859,13 +859,25 @@ + (void)showReviewController {
...
+        [[NSUserDefaults standardUserDefaults] setObject:currentVersion forKey:@"VersionUnderReview"];
+        [[NSUserDefaults standardUserDefaults] setInteger:0 forKey:@"PlaybackAttempts"];
+    }
+    else {
+        // Read and increase count by 1
+        NSInteger count = [[NSUserDefaults standardUserDefaults] integerForKey:@"PlaybackAttempts"] + 1;
+        [[NSUserDefaults standardUserDefaults] setInteger:count forKey:@"PlaybackAttempts"];
+        
+        // Show review popup each 10th launch/wakeup
+        if (count % 50 == 0) {
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:878]: either code or comment is incorrect

File: XBMC Remote/Utilities.m [hunk 3]
```diff
@@ -859,13 +859,25 @@ + (void)showReviewController {
...
-        [Utilities showReviewController];
+    id infoDict = NSBundle.mainBundle.infoDictionary;
+    NSString *currentVersion = [NSString stringWithFormat:@"v%@ (%@)", infoDict[@"CFBundleShortVersionString"],
+                                                                       infoDict[(NSString*)kCFBundleVersionKey]];
+    NSString *savedVersion = [[NSUserDefaults standardUserDefaults] stringForKey:@"VersionUnderReview"];
+    // Compare current version with version under review
+    if (![savedVersion isEqualToString:currentVersion]) {
+        // Reset counter to 0 for new version
+        [[NSUserDefaults standardUserDefaults] setObject:currentVersion forKey:@"VersionUnderReview"];
+        [[NSUserDefaults standardUserDefaults] setInteger:0 forKey:@"PlaybackAttempts"];
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:870]: please don't use "magic" strings

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

**Comment by kambala-decapitator** [comment]
> @kambala-decapitator, do you see this is a good choice to trigger the reviews? And if so, should we add this for the next release already?

let's do it!

File: XBMC Remote/Utilities.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m:862]: Will use NSDictionary.

File: XBMC Remote/Utilities.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m:878]: Corrected the comment.

File: XBMC Remote/Utilities.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m]: Used defines now.

**Review by kambala-decapitator — APPROVED** [review]

**Comment by wutschel** [comment]
Included one more fix (comment still referred to "50th launch/wakeup", but the trigger comes after "50th playback attempt"). Squashed all changes into one and updated the commit message.

File: XBMC Remote/Utilities.h [hunk 4]
```diff
@@ -88,5 +88,7 @@ typedef enum {
 + (BOOL)hasRemoteToolBar;
 + (CGFloat)getBottomPadding;
 + (void)sendXbmcHttp:(NSString*)command;
++ (void)showReviewController;
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.h:91]: not sure why this method is public

Comment (wutschel) [reply XBMC Remote/Utilities.h]: A remnant of the first implementations I did. I now removed it as part of the latest forced push.

**Comment by wutschel** [comment]
Updated with last request for change.

**Pull request merged**